full/src/test/java/apoc/systemdb/SystemDbTest.java

```diff
@@ -74,5 +103,77 @@ public void testWriteStatements() {
+        String declareStatementFromProcedure = "CALL apoc.custom.declareProcedure('procName(input = 42 :: INTEGER?) :: (answer :: NUMBER?)', 'RETURN $input as answer', 'READ', 'Procedure that answer to the Ultimate Question of Life, the Universe, and Everything');";
 
+        // We test uuid, we also need to export the related constraint (in another file)
+        final String constraintForUuid = "CREATE CONSTRAINT IF NOT EXISTS ON (n:Person) ASSERT n.alpha IS UNIQUE;";
+        db.executeTransactionally(constraintForUuid);
+        final String uuidStatement = "CALL apoc.uuid.install('Person', {uuidProperty:\"alpha\",addToSetLabels:true});";
+        db.executeTransactionally(uuidStatement);
+
+        // We test the data virtualization catalog
+        final String dvStatement = "CALL apoc.dv.catalog.add('dvName', {name:\"dvName\",url:\"file://myUrl\",desc:\"person's details\",labels:[\"Person\"],query:\"map.name = $name and map.age = $age\",params:[\"$name\",\"$age\"],type:\"CSV\"})";
+        db.executeTransactionally(dvStatement);
+
+        TestUtil.testCall(db, "CALL apoc.systemdb.export.metadata", row -> {
+            assertEquals(8L, row.get("rows"));
+            assertEquals(true, row.get("done"));
+            assertEquals("metadata", row.get("file"));
+        });
+        
+        assertEquals(Set.of(constraintForUuid), readLines("metadata.Uuid.schema.neo4j.cypher"));
+        assertEquals(Set.of(uuidStatement), readLines("metadata.Uuid.neo4j.cypher"));
+        assertEquals(Set.of(triggerOne, triggerTwo, pauseTrigger), readLines("metadata.Trigger.neo4j.cypher"));
+        assertEquals(Set.of(declareProcedure, declareStatementFromProcedure), readLines("metadata.CypherProcedure.neo4j.cypher"));
+        assertEquals(Set.of(declareFunction, declareStatementFromFunction), readLines("metadata.CypherFunction.neo4j.cypher"));
+        assertEquals(Set.of(dvStatement), readLines("metadata.DataVirtualizationCatalog.neo4j.cypher"));
+        
+        // -- with config and uuid constrain dropped
+        db.executeTransactionally("DROP CONSTRAINT ON (p:Person) ASSERT p.alpha IS UNIQUE");
+        TestUtil.testCall(db, "CALL apoc.systemdb.export.metadata($config)", 
+                Map.of("config", Map.of(FILENAME_KEY, "custom", FEATURES_KEY, Set.of(ExportMetadata.Type.Uuid.name()))), 
+                row -> {
+                    assertEquals(1L, row.get("rows"));
+                    assertEquals(true, row.get("done"));
+                    assertEquals("custom", row.get("file")); 
+        });
+
+        db.executeTransactionally("CALL apoc.uuid.removeAll");
+        db.executeTransactionally("CALL apoc.trigger.removeAll");
+
+        assertEquals(Set.of(constraintForUuid), readLines("custom.Uuid.schema.neo4j.cypher"));
+        assertEquals(Set.of(uuidStatement), readLines("custom.Uuid.neo4j.cypher"));
+    }
+
+    private Set<String> readLines(String fileName) {
```

Comment:
se cerchi nel codice dovrebbe già esserci, in caso negativo mettilo dentro TestUtil